src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx

```diff
@@ -273,128 +402,159 @@ const DataSeriesFormItemModal = ({
+          <div className={`${baseClassName}--input-group--item`}>
+            <TextInput
+              id={`${id}_attribute-unit`}
+              labelText={mergedI18n.dataItemEditorDataItemUnit}
+              light
+              placeholder={`${mergedI18n.example}: %`}
+              onChange={(evt) =>
                 setEditDataItem({
                   ...editDataItem,
-                  dataFilter,
-                });
-              } else {
-                setEditDataItem({
-                  ...omit(editDataItem, 'dataFilter'),
-                });
+                  unit: evt.target.value,
+                })
               }
-            }}
-            titleText={mergedI18n.dataItemEditorDataItemFilter}
-          />
+              value={editDataItem.unit}
+            />
+          </div>
+          {type === CARD_TYPES.VALUE && (
+            <div className={`${baseClassName}--input-group--item-end`}>
+              <Dropdown
+                id={`${id}_value-card-decimal-place`}
+                titleText="Decimal places"
+                direction="bottom"
+                label=""
+                items={[mergedI18n.notSet, '0', '1', '2', '3', '4']}
+                light
+                selectedItem={
+                  editDataItem.precision?.toString() || mergedI18n.notSet
+                }
+                onChange={({ selectedItem }) => {
+                  const isSet = selectedItem !== mergedI18n.notSet;
+                  if (isSet) {
+                    setEditDataItem({
+                      ...editDataItem,
+                      precision: Number(selectedItem),
+                    });
+                  } else {
+                    setEditDataItem(omit(editDataItem, 'precision'));
+                  }
+                }}
+              />
+            </div>
+          )}
         </div>
-        {!isEmpty(editDataItem.dataFilter) &&
-        availableDimensions[selectedDimensionFilter] ? (
-          <div className={`${baseClassName}--input-group--item-end`}>
+      )}
+
+      {isSummaryDashboard ? ( // only show data filter in summary dashboards
+        <div className={`${baseClassName}--input-group`}>
+          <div
+            className={classnames({
+              [`${baseClassName}--input-group--item`]: !isEmpty(
+                editDataItem.dataFilter
+              ),
+              [`${baseClassName}--input-group--item-half`]:
+                isEmpty(editDataItem.dataFilter) ||
+                (!isEmpty(editDataItem.dataFilter) &&
+                  !availableDimensions[selectedDimensionFilter]),
+            })}>
             <Dropdown
-              id={`${id}_data-filter-value`}
+              id={`${id}_data-filter-key`}
               label=""
               translateWithId={handleTranslation}
               direction="bottom"
-              items={availableDimensions[selectedDimensionFilter]}
+              items={[mergedI18n.none, ...Object.keys(availableDimensions)]}
```

Comment:
I've seen `mergedI18n.xxxx` be used as selectable items a lot which could cause issues when the user saves the dashboard. I think its good to translate the labels, but the underlying item that gets selected should always be the same so that we never have to worry about locales in the underlying data

I may be reading it wrong though

Reply:
I have updated the dropdowns that would place the selected item into the json payload (the grains dropdown for example) to use the `{ id: '', text: '' }` convention so that the user can specify their own translation while we reference a source of truth with the `id`. Note: this required an update to the `aggregationMethods` property on the `dataItems` proptype to accomplish.